### ISSUE TITLE
Add more automation to projects so that issues moved into 'Done' are marked as closed

### ISSUE THREAD

**Issue by jackchmbrln** [dear-github/dear-github#284]
I love how closing an issue automatically moves it into my 'Done' column in the project but...I would love it even more if we could also have it the other way around—so that moving an issue into the 'Done' column would mark it as closed

**Comment by adamdry** [comment]
Couldn't agree more. I drive my day from the Project drag and drop board, moving an Issue to Done should be the only thing I need to do to mark it as done.

**Comment by marcelstoer** [comment]
Fully agree. Was very surprised when I found out this isn't done automatically. It's like the sanest thing to expect, isn't it?

**Comment by LucasHarvey** [comment]
I agree as well! Encountered this today and I wish it was automatic instead of looking through my many issues and closing the ones that are done.

**Comment by marcelstoer** [comment]
This is what I got from GH Support two weeks ago (same as always):

> I agree that would be useful, and it is certainly something that we're considering on our end. I will add a +1 for you to our internal feature request list, although I wouldn't be able to promise that this feature will be added in the future.

**Comment by agentlewis** [comment]
Maybe we should get the microsoft ux team involved :smiley_cat:

**Comment by dessant** [comment]
I've just published a GitHub App that closes issues which are moved to the Done column. :tada: 

https://github.com/dessant/close-done-issues

Let me know how it works for you!

**Comment by jackchmbrln** [comment]
@dessant Wow, thanks this is great! Would there be any way of making it so that moving an issue back into to 'In progress' or 'To-do' would reopen it?

**Comment by dessant** [comment]
@jackchmbrln, I like the idea, though once it's more than a single action, people will want different behaviors. :yum: A config option could be added to enable a list of columns that reopen issues moved there.

It would look like the config in this app:
https://github.com/dessant/move-issues

**Comment by jackchmbrln** [comment]
Yes, true I suppose it could get tricky with everyone likely having different names for the columns, a config file would work well! That bot is just what I needed though, thank you!

**Comment by dessant** [comment]
Here's a reworked version of the app that can be configured, it supports opening and closing issues depending on the project column.

https://github.com/dessant/issue-states

**Comment by adamdry** [comment]
I haven't tried it but it looks like it'll do the job. Nice one!!

**Comment by saerdnaer** [comment]
See also: https://github.com/philschatz/project-bot

**Comment by eL0ck** [comment]
👍

**Comment by francos** [comment]
Thanks for this app @dessant! 

I was wondering if it works with projects created at the user or organization level rather than at the repository level as I installed it and it didn't seem to work for me.

**Comment by padcom** [comment]
Is there any movement in this topic? It's 2021 and the project page still doesn't allow for automation of the ticket status after moving it to the Done column... Guys, take it seriously, please.

**Comment by ianderrington** [comment]
+1 from my end. Thanks!

**Comment by vojtechsanda** [comment]
+1 From me as well

**Comment by gflarity** [comment]
Projects have potential but things like this lead me to believe it's not really a serious feature.

**Comment by EhrenEschmann** [comment]
+1 this would be very helpful.

**Comment by hdrhmd** [comment]
Still open?

+1 to this feature

**Comment by Devgasparotto** [comment]
This is essential.

+1

**Comment by spalfy** [comment]
+1 absolutely essential to have a way to set this up via the GUI

**Comment by gabrieleolmi** [comment]
+1 essential feature!!!

**Comment by MartinX3** [comment]
Look at new projects beta
We got workflows.

**Comment by gabrieleolmi** [comment]
> Look at new projects beta We got workflows.

Workflows cannot be edited. It says "coming soon feature".

**Comment by MartinX3** [comment]
Yep, so we can close this "issue"

**Comment by jasonaycock** [comment]
I wouldn't close it until it works.  Right now the idea is there as part of the beta, but it's not implemented.

**Comment by LukasFelzmann** [comment]
+1 very important feature

**Comment by SimiHunjan** [comment]
+1 very important feature

**Comment by TanguySoto** [comment]
+1

**Comment by dasmerlon** [comment]
For those still having the problem, I've created a [GitHub Action](https://github.com/dasmerlon/project-issue-state-sync) that closes and reopens issues if they are in specific project columns. 

Unfortunately, the workflow cannot be triggered when an issue is moved to a different project column, as there's no event one can subscribe to. But you can schedule it so that the issue states are updated regularly. :)

**Comment by joshua-reddish** [comment]
+1

**Comment by ccrowley96** [comment]
+10

**Comment by alexhajdu** [comment]
+1 
How this is not implemented yet?

**Comment by Benjamin-Justice** [comment]
+1
Working with Github projects with this behaviour is very frustrating.
if I use Github Projects, I cannot use Issues, as Projects will SPAM issues into an unusable state with 90% dead tickets.

"It's free" is currently the only reason a small project of mine is using it.

**Comment by 0xRoch** [comment]
+1

**Comment by jmicho-alx** [comment]
+1

**Comment by artnikbrothers** [comment]
+1

**Comment by poponealex** [comment]
+1
A must need

**Comment by alexthe2** [comment]
It's 2023 how is this still not a feature?

**Comment by cargopete** [comment]
Please add this folks

**Comment by aborruso** [comment]
+1
A must need

**Comment by jthantranon** [comment]
+1

**Comment by Nasfame** [comment]
Not added in 2023: i guess its never going to be

**Comment by johnhungerford** [comment]
+1

**Comment by dmrickey** [comment]
Obviously sane requests have fallen on deaf ears. It _might_ be a good product (jury is still out on that one), but this is the last "little thing" that's just going to keep me from adopting Github Issues.

Let's see, this was created 5 1/2 years ago? I guess I'll check back 2029 to see how it's progressed ¯\\\_(ツ)_/¯

**Comment by igor-korotenko** [comment]
**Dear** Github Team, please do smth with this, you are clearly ignoring absolutely must-have feature-request, thats just 🤦‍♂️

**Comment by jsolly** [comment]
This is SO annoying. I swear it adds 30 seconds to our dev standups because I have to individually open up each issue placed in the `done` column to close it.

**Comment by mindaugasrukas** [comment]
+1

**Comment by artgonline** [comment]
so, just let me get it straight... it's been almost 6 years since this issues was opened and this functionality doesn't exist yet? wow...

**Comment by gtirloni** [comment]
I hate to add +1 comments to issues but this is clearly absurd. This is the kind of thing I'd expect from Atlassian (see [here](https://jira.atlassian.com/browse/JRASERVER-14543?filter=98704&jql=project%20%3D%20JRASERVER%20AND%20status%20in%20(Open%2C%20%22In%20Progress%22%2C%20%22Needs%20Triage%22)%20AND%20resolution%20%3D%20Unresolved%20ORDER%20BY%20created%20ASC%2C%20priority%20DESC%2C%20updated%20DESC), not a competitor.

**Comment by Kendja** [comment]
It would be nice. Was surprised it's not the feature already.
Since we have "convert to issue"

**Comment by Noctunus** [comment]
Yet another +1. Just migrated over from Atlassian to GitHub and this is the only feature which is missing.

**Comment by ethanstrominger** [comment]
+1

**Comment by 4cyc** [comment]
So many useful features GitHub provides with GitHub Projects, so surprised this obvious one is missing.

+1

2024

**Comment by LadyBanks** [comment]
Agreed with everyone.  This is such an obvious need.

**Comment by mukteshkrmishra** [comment]
+1

**Comment by thane-eisener** [comment]
Started using github projects a couple of months ago. This is a major flaw. It's been 6 years (!) today since this was raised

**Comment by nickponomar** [comment]
+1
Why is that not a feature still!?

**Comment by gunhaxxor** [comment]
+1

**Comment by dovh-me** [comment]
+1

**Comment by janlis-ff** [comment]
+1

**Comment by LoganMalcome** [comment]
+1

**Comment by MeyerJoey** [comment]
![travis-neil-primrose](https://github.com/dear-github/dear-github/assets/7315193/df8a8a75-43af-45af-915a-b0afeb0b6498)

**Comment by franklinkim** [comment]
+1

**Comment by AdrianVaduva4k** [comment]
Haha +1

**Comment by BennyHoang** [comment]
+1

**Comment by lekipising** [comment]
+1

**Comment by mwithi** [comment]
+1

**Comment by mistermark** [comment]
How about it, folks? Are we getting this?

**Comment by ReneTC** [comment]
This would be a great feature for "workflows".

**Comment by BennyHoang** [comment]
> How about it, folks? Are we getting this?

this issue was submitted in 2018 🤷

**Comment by Waterstraal** [comment]
GitHub team, is there a reason why this hasn't been implemented yet? Looks like an easy automation to add to project workflows?

**Comment by JoshDLane** [comment]
+1

**Comment by valentinochet** [comment]
+1

**Comment by aliridha1510** [comment]
+1

**Comment by HatemMn** [comment]
+1

**Comment by matart15** [comment]
+1

**Comment by Elvin1492** [comment]
+100000

**Comment by BennyHoang** [comment]
It is out now! https://github.blog/changelog/2024-04-25-github-issues-projects-auto-close-issue-project-workflow/

However now if you are pulling issues from done back, you need to remember to reopen issue 👀

**Comment by Nukesor** [comment]
Stumbled upon the same issue as @BennyHoang. The new workflow isn't implemented in a generic way, but only to handle transitions from `Open` -> `Closed`.

The [Project issue state sync](https://github.com/dasmerlon/project-issue-state-sync) Github Action still takes care of that though, so I guess I'll keep using it for the time being.

**Comment by sovetski** [comment]
Thank you @BennyHoang for this great news!
